### PR TITLE
Changes needed to deploy SRO from OperatorHub as a community operator

### PR DESCRIPTION
**Pull request by dagrayvid**
The changes to the operator include:
 - Create ClusterOperator if not found (as it can't be deployed as part of the bundle when deploying with OLM)
 - Fixing two typos in controllers/status.go

The new config/manifests is created by the operator-sdk, as a skeleton for the CSV that will be created when running `make bundle`. This will be useful in the future when we are updating the bundle, we won't need to set these fields again (with the exception of alm-examples, for some reason this cannot be set in the skeleton CSV).

Once this PR is merged I will create a pr to update 
https://github.com/operator-framework/community-operators/blob/master/community-operators/special-resource-operator/0.0.1/special-resource-operator.v0.0.1.clusterserviceversion.yaml

Some potentially helpful links:
- https://operator-framework.github.io/community-operators/testing-operators/#preparing-file-structure
- https://sdk.operatorframework.io/docs/olm-integration/generation/#csv-fields
- https://sdk.operatorframework.io/docs/olm-integration/quickstart-bundle/
- https://github.com/operator-framework/operator-registry/blob/master/docs/design/operator-bundle.md

### CODE REVIEW

File: config/manifests/bases/special-resource-operator.clusterserviceversion.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    type: AllNamespaces
+  keywords:
+  - GPI
+  links:
+  - name: README
+    url: https://github.com/openshift-psap/special-resource-operator
+  maintainers:
+  - email: zkaiser@redhat.com
+    name: Zvonko Kaiser
+  maturity: alpha
```
Comment (zvonkok) [inline config/manifests/bases/special-resource-operator.clusterserviceversion.yaml:55]:
```suggestion
  - email: dagray@redhat.com
    name: David Gray
  maturity: alpha
```

File: config/manifests/bases/special-resource-operator.clusterserviceversion.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  provider:
+    name: Red Hat
+    url: redhat.com
+  version: 0.0.0
```
Comment (zvonkok) [inline config/manifests/bases/special-resource-operator.clusterserviceversion.yaml:59]: We need to set here something right?

File: controllers/status.go [hunk 3]
```diff
@@ -50,13 +50,22 @@ func (r *SpecialResourceReconciler) clusterOperatorStatusGetOrCreate() error {
...
+
+	// If we land here there is no clusteroperator object for SRO, create it.
+	log = r.Log.WithName(color.Print("status", color.Blue))
+	log.Info("No ClusterOperator found... Creating ClusterOperator for SRO")
+
+	co := &configv1.ClusterOperator{ObjectMeta: metav1.ObjectMeta{Name: r.GetName()}}
+
+	co, err = r.ClusterOperators().Create(context.TODO(), co, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to create clusteroperator %s: %v", co.Name, err)
```
Comment (zvonkok) [inline controllers/status.go]:
```suggestion
		return errs.Wrap(err, "Failed to create ClusterOperator " + co.Name)
```

File: config/manifests/bases/special-resource-operator.clusterserviceversion.yaml (hunk 2 above)
Comment (dagrayvid) [reply config/manifests/bases/special-resource-operator.clusterserviceversion.yaml:59]: The version will be overwritten by the VERSION in the makefile when we `make bundle`

**Pull request merged**